### PR TITLE
Support for `_path` and `_file` virtual columns for table function `url`.

### LINKED ISSUE #49231: Support for `_path` and `_file` virtual columns for table function `url`.
```
play-eu :) SELECT _file, count() FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-00{00..59}00.gz', LineAsString) GROUP BY _file ORDER BY _file

SELECT
    _file,
    count()
FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-00{00..59}00.gz', LineAsString)
GROUP BY _file
ORDER BY _file ASC

Query id: a208a18c-02f9-47cc-8848-a6895bba3b49


0 rows in set. Elapsed: 0.023 sec. 

Received exception from server (version 23.4.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: '_file' while processing query: 'SELECT _file, count() FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-00{00..59}00.gz', 'LineAsString') GROUP BY _file ORDER BY _file ASC', required columns: '_file'. (UNKNOWN_IDENTIFIER)

play-eu :) SELECT _path, count() FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-00{00..59}00.gz', LineAsString) GROUP BY _path ORDER BY _path

SELECT
    _path,
    count()
FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-00{00..59}00.gz', LineAsString)
GROUP BY _path
ORDER BY _path ASC

Query id: 495fdc97-0069-4da7-b2ac-af015fb45f25


0 rows in set. Elapsed: 0.025 sec. 

Received exception from server (version 23.4.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: '_path' while processing query: 'SELECT _path, count() FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-00{00..59}00.gz', 'LineAsString') GROUP BY _path ORDER BY _path ASC', required columns: '_path'. (UNKNOWN_IDENTIFIER)
```

### LINKED ISSUE #49232: The limit on the maximum number of generated addresses cannot be changed for function `url`
```
play-eu :) SELECT _path, count() FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-{00..23}{00..59}00.gz', LineAsString) GROUP BY _path ORDER BY _path

SELECT
    _path,
    count()
FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-{00..23}{00..59}00.gz', LineAsString)
GROUP BY _path
ORDER BY _path ASC

Query id: 90cd5580-a567-4af1-9b6f-ea99035053b1


0 rows in set. Elapsed: 0.022 sec. 

Received exception from server (version 23.4.1):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Table function 'remote': first argument generates too many result addresses. (BAD_ARGUMENTS)

play-eu :) SET table_function_remote_max_addresses = 1000000

SET table_function_remote_max_addresses = 1000000

Query id: 31702a49-0a8b-4580-88fc-a87ef2079764

Ok.

0 rows in set. Elapsed: 0.001 sec. 

play-eu :) SELECT _path, count() FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-{00..23}{00..59}00.gz', LineAsString) GROUP BY _path ORDER BY _path

SELECT
    _path,
    count()
FROM url('https://clickhouse-public-datasets.s3.amazonaws.com/wikistat/original/pageviews-20200101-{00..23}{00..59}00.gz', LineAsString)
GROUP BY _path
ORDER BY _path ASC

Query id: 1e18851a-f9d7-43e4-84ef-3ffd34fb4b14


0 rows in set. Elapsed: 0.001 sec. 

Received exception from server (version 23.4.1):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Table function 'remote': first argument generates too many result addresses. (BAD_ARGUMENTS)
```

Also, take a look at the exception message. It is wrong.

### PR DESCRIPTION
**Pull request by Ziy1-Tan**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Add virtual column `_file` and `_path` support to table function `url`. 
- Impove error message for table function `url`. 
- resolves #49231
- resolves #49232

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 2c159061edce46adf9dfd83a82129cb68ed1a9f3 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49356/2c159061edce46adf9dfd83a82129cb68ed1a9f3/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

File: src/Common/parseRemoteDescription.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ static bool parseNumber(const String & description, size_t l, size_t r, size_t &
  * abc{1..9}de{f,g,h}   - is a direct product, 27 shards.
  * abc{1..9}de{0|1}     - is a direct product, 9 shards, in each 2 replicas.
  */
-std::vector<String> parseRemoteDescription(const String & description, size_t l, size_t r, char separator, size_t max_addresses)
+std::vector<String> parseRemoteDescription(const String & description, size_t l, size_t r, char separator, size_t max_addresses, const String& func_name)
```
Comment (ucasfl) [inline src/Common/parseRemoteDescription.cpp:67]: Format:
```suggestion
std::vector<String> parseRemoteDescription(const String & description, size_t l, size_t r, char separator, size_t max_addresses, const String & func_name)
```

**Comment by Ziy1-Tan** [comment]
@evillique @alexey-milovidov  mind take a look? :)

**Comment by Ziy1-Tan** [comment]
@alexey-milovidov 
Hi! It seems that the failing test is not related to my changes, can you get someone to check it?
- 01646_rewrite_sum_if_bug 
- 01600_parts_types_metrics_long
- 01600_parts_states_metrics_long
- 00992_system_parts_race_condition_zookeeper_long
- 00002_log_and_exception_messages_formatting

**Review by evillique — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@evillique, the example from the issue does not work at all.

**Pull request merged**